vault/resource_plugin_pinned_version.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func pluginPinnedVersionResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: provider.MountCreateContextWrapper(pluginPinnedVersionWrite, provider.VaultVersion116),
+		UpdateContext: provider.UpdateContextWrapper(pluginPinnedVersionWrite, provider.VaultVersion116),
+		ReadContext:   pluginPinnedVersionRead,
+		DeleteContext: pluginPinnedVersionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldType: {
+				Type:        schema.TypeString,
+				Description: `Type of plugin; one of "auth", "secret", or "database".`,
+				Required:    true,
+				ForceNew:    true,
+			},
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Description: "Name of the plugin.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			consts.FieldVersion: {
+				Type:        schema.TypeString,
+				Description: "Semantic pinned plugin version.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func pluginPinnedVersionWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := fmt.Sprintf("%s/%s", d.Get(consts.FieldType).(string), d.Get(consts.FieldName).(string))
+
+	log.Printf("[DEBUG] Writing pinned plugin version %q", id)
+	_, err = client.Logical().WriteWithContext(ctx, idToPath(id), map[string]interface{}{
+		"version": d.Get(consts.FieldVersion).(string),
+	})
+	if err != nil {
+		return diag.Errorf("error updating pinned plugin version %q: %s", id, err)
+	}
+	log.Printf("[DEBUG] Wrote pinned plugin version %q", id)
+
+	d.SetId(id)
+
+	return nil
```

Comment:
similar to above, should we read after writes here?
```suggestion
	return pluginPinnedVersionRead(ctx, d, meta)
```